### PR TITLE
Fix PHP4 style constructs

### PR DESCRIPTION
**Pull request by Scott-Mc**
Fixing the old PHP4 style constructs across the various drivers for PHP7 comparability.

I left the mysql and mssql drivers as is given these are both depreciated.

### CODE REVIEW

**Comment by jv2222** [comment]
This is an interesting scenario. A lot of good (I think) changes here but the lib is so old and no test suite. Also, I haven't personally worked with the library for a few years. I could just click merge but I'm not 100% sure that it won't cause issues.

In a bit of a pickle and not sure how to move forward!

I guess we could go through each change line by line and discuss with @gnowxilef and other maintainers.

I think this really points to the need for a unit test suite but I don't have the bandwidth for that :(

**Comment by Scott-Mc** [comment]
I've only been able to test mysqli,  sqlite and postgres and all seem fine on both PHP 5.6 and PHP 7   The bulk of the changes is just moving from the old way to declare constructs to avoid logs being flooded.  I intentionally left out the mysql/mssql driver given these are depreciated in newer versions so wouldn't work anyway for those still running the old version.

I've also tested the set names and this works perfectly fine.

A tested suite would be nice

**Comment by jv2222** [comment]
@gnowxilef What do you recommend here?

**Comment by gnowxilef** [comment]
i agree with @jv2222. it would be great to have a test suite, but that is a big undertaking. (i think that is already a github issue.) it seems like it was tested to the best possible, so i'm fine with it.

File: mysqli/ez_sql_mysqli.php [hunk 1]
```diff
@@ -161,7 +161,7 @@ function select($dbname='', $encoding='')
 						$charsets[] = $row["Charset"];
 					}
 					if(in_array($encoding,$charsets)){
-						$this->dbh->query("SET NAMES '".$encoding."'");						
+						$this->dbh->set_charset($encoding);
```
Comment (gnowxilef) [inline mysqli/ez_sql_mysqli.php]: one question though: where is `set_charset` defined?

**Comment by gnowxilef** [comment]
fixes #102

**Comment by gnowxilef** [comment]
should also fix #117

File: mysqli/ez_sql_mysqli.php (hunk 1 above)
Comment (Scott-Mc) [inline mysqli/ez_sql_mysqli.php]: Do you mean this? http://nl3.php.net/manual/en/mysqli.set-charset.php

**Comment by surruk51** [comment]
There are a lot of vars in the orginal - which imply public access on variables that clearly are really for internal use only. There is also the opportunity to catch buggy input in __set and __get routines. There are opportunities to make it better, not just to change the syntax to the PHP5/7 way of making classes. But I guess this would be a fork.

**Pull request merged**